gcp/apis/compute/v1alpha1/globaladdress_types.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// GlobalAddressSpec defines the desired state of a GlobalAddress
+type GlobalAddressSpec struct {
+	v1alpha1.ResourceSpec   `json:",inline"`
+	GlobalAddressParameters `json:",inline"`
+}
+
+// GlobalAddressParameters specifies the configuration of a GlobalAddress.
+type GlobalAddressParameters struct {
+	// Address: The static IP address represented by this resource.
+	// +optional
+	Address *string `json:"address,omitempty"`
+
+	// AddressType: The type of address to reserve, either INTERNAL or
+	// EXTERNAL. If unspecified, defaults to EXTERNAL.
+	//
+	// Possible values:
+	//   "EXTERNAL"
+	//   "INTERNAL"
+	//   "UNSPECIFIED_TYPE"
+	// +optional
+	AddressType *string `json:"addressType,omitempty"`
+
+	// Description: An optional description of this resource.
+	// +optional
+	Description *string `json:"description,omitempty"`
+
+	// IPVersion: The IP version that will be used by this address. Valid
+	// options are IPV4 or IPV6.
+	//
+	// Possible values:
+	//   "IPV4"
```

Comment:
Not related to this PR specifically. But I think at some point, we should have a validation mechanism for given possible values. Opened https://github.com/crossplaneio/stack-gcp/issues/12